File: client/makc2021-react-redux-ts/src/Layer4/Domains/DummyMain/Queries/List/Get/DomainListGetQueryOutput.ts
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

import {
  createListGetQueryOutput,
  ListGetQueryOutput
} from 'src/Layer2/Queries/List/Get/ListGetQueryOutput';
import { DomainItemGetQueryOutput } from '../../Item/Get/DomainItemGetQueryOutput';

/**
 * Выходные данные запроса на получение списка в домене.
 */
export interface DomainListGetQueryOutput
  extends ListGetQueryOutput<DomainItemGetQueryOutput> {}

/**
 * Создать выходные данные запроса на получение списка в домене.
 * @returns Выходные данные запроса на получение списка в домене.
 */
export function createDomainListGetQueryOutput() {
  return {
    ...createListGetQueryOutput<DomainItemGetQueryOutput>()
  } as DomainListGetQueryOutput;
}

File: server/src/Makc2021.Layer4.Sql.Domains.DummyMain/Queries/List/Get/DomainListGetQueryOutput.cs
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Makc2021.Layer2.Sql.Queries.List.Get;
using Makc2021.Layer4.Sql.Domains.DummyMain.Queries.Item.Get;

namespace Makc2021.Layer4.Sql.Domains.DummyMain.Queries.List.Get
{
    /// <summary>
    /// Выходные данные запроса на получение списка в домене.
    /// </summary>
    public class DomainListGetQueryOutput : ListGetQueryOutput<DomainItemGetQueryOutput>
    {
    }
}
